AsyncDisplayKit/ASImageNode.mm

```diff
@@ -271,8 +285,9 @@ - (void)displayDidFinish
 {
   [super displayDidFinish];
 
+  ASDN::MutexLocker l(_imageLock);
   // If we've got a block to perform after displaying, do it.
-  if (self.image && _displayCompletionBlock) {
+  if (_image && _displayCompletionBlock) {
 
     // FIXME: _displayCompletionBlock is not protected by lock
     _displayCompletionBlock(NO);
```

Comment:
It is likely essential that we release the lock before calling this block. We may want to access the block pointer and store it in a local variable before calling it, which could be a plausible threading issue if the display completion block is changed in the case that a display is canceled or restarted while one is in progress.